app/views/RoomsListView/index.js

```diff
@@ -84,11 +82,12 @@ const shouldUpdateProps = [
 	'searchText',
 	'loadingServer',
 	'showServerDropdown',
-	'showSortDropdown',
 	'sortBy',
 	'groupByType',
 	'showFavorites',
 	'showUnread',
+	'showAvatar',
+	'displayType',
```

Comment:
For some reason it takes a couple of seconds (10-15) to update the `RoomsListView` after updating the sorting preference, might be related to how this screen is being updated

Reply:
https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/3318/commits/d6d460b37b43b1caff7b15e02e77c86e0d93fd57